### DISCUSSION TITLE
When clicked on Slider, tooltip doesnt get cleared even if widget tab is switched

### DISCUSSION THREAD

**Discussion by muthubentley** [iTwin/iTwinUI#1122] [Q&A] [answered]
When clicked on Slider, tooltip doesnt  get cleared even if widget tab is switched
When clicked on Slider:
![image](https://user-images.githubusercontent.com/107229931/223722438-02ef11db-14a3-42ce-a1bf-73462fa98117.png)

When switching the widget tab:
![image](https://user-images.githubusercontent.com/107229931/223722637-3bdf30c2-5c80-4578-ab66-282d8bea8557.png)

This behavior can be reproduced in this sandbox
https://www.itwinjs.org/sandbox/MuthuKalamani/ElevationChange

I think focus is not getting cleared even if the tab is switched. Please let us know the way to clear the tooltip

**Comment by elephantcatdog** [comment]
Hello @muthubentley,
I have two solutions for you:
1. Use `tooltipProps` to set `visible` to true and false when you switch tabs.
2. Use the same `tooltipProps` to remove the tooltip entirely. I only suggest this as an option because you have the result of the slider in the 'Elevation(in metre):' input box below.

Here's how you can implement that, where `false` can also be a variable to keep track of its visibility:
![image](https://user-images.githubusercontent.com/37936169/223774771-4d26e90b-9b21-4f11-a00b-37f9ee290b95.png)
Here's an example on our storybook: 
https://itwin.github.io/iTwinUI/react/?path=/story/input-slider--custom-tick-no-tooltip

**Reply by muthubentley** [reply]
Hi @elephantcatdog 

Thanks for the workarounds. Actually the sandbox I sent is a sample. In the actual application result of slider is not shown. So the tooltip might be required. 
Coming to the rootcause of the issue ,shouldnt the tooltip get cleared when switching tabs. This seems to be a bug to me? Can it be resolved

**Comment by mayank99** [comment, marked as answer]
Hey @muthubentley, this is actually a bug in iTwin.js. I opened an issue in appui if you want to track that: https://github.com/iTwin/appui/issues/195

**Reply by muthubentley** [reply]
Thanks @mayank99  I will track the issue.

**Comment by raplemie** [comment]
@mayank99 I'll look into the tab not getting focus, but I think the tooltip of the slider should go away when the Slider is unmounted, regardless of the focus. (IE, in the provided sandbox, if you go to the empty widget and return to the slider widget, and click on the "elevation(in meter)", you'll notice that the tooltip will remain in place, regardless of the focus.

**Reply by mayank99** [reply]
I think you're right, although I swear I remember seeing slider was still mounted and focus was still on it when I was initially looking into it.

In either case, I have created https://github.com/iTwin/iTwinUI/pull/1139 which should fix this issue. Do you know what would be the best way to test this?

**Reply by mayank99** [reply]
update: tested these two cases in the iTwin.js sandbox and the second tooltip correctly disappears when switching widgets whereas the first one does not. 

```jsx
<Tooltip content='im a tooltip!' visible={tooltipVisible}>
  <button
    onFocus={() => setTooltipVisible(true)}
    onBlur={() => setTooltipVisible(false)}>
     not good
  </button>
</Tooltip>
```
```jsx
<Tooltip content='im also a tooltip!' trigger='mouseenter focus click'>
  <button>good</button>
</Tooltip>
```
so #1139 should work fine.

**Reply by muthubentley** [reply]
@mayank99  is this issue fixed and released. Can you please let me know the version of package so that I could test before conveying to the partner.

**Reply by mayank99** [reply]
it's not yet released but will likely be in `@itwin/itwinui-react@2.7.0` which will be available in the next few days.

**Reply by muthubentley** [reply]
@mayank99  is this issue released in @itwin/itwinui-react@2.7.0 ?

**Reply by mayank99** [reply]
yes

**Reply by muthubentley** [reply]
Thanks for the confirmation @mayank99
